components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RDBMSSessionDataStore.java

```diff
@@ -346,25 +247,62 @@ public void persistSessionData(String key, String type, Object entry, long nanoT
             return;
         }
         PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+
         try {
-            String sqlQuery = getSessionStoreDBQuery(sqlInsertSTORE, type);
-            preparedStatement = connection.prepareStatement(sqlQuery);
+            if (StringUtils.isBlank(sqlSelect)) {
```

Comment:
Why do we need to check the blank status? Anyway, we need to set the query right? If that's the case we can omit this if condition.